packages/material-ui-unstyled/src/composeClasses/composeClasses.ts

```diff
@@ -0,0 +1,27 @@
+export default function composeClasses<ClassKey extends string>(
+  slots: Record<ClassKey, Array<string | false>>,
+  getUtilityClass: (slot: string) => string,
+  classes?: Record<string, string>,
```

Comment:
@mnajdova When is this ok be optional? Every usage is passing `classes` so it seems to me we can mark this as requred. And then the next question would be if this can actually be `undefined` or if the caller should make sure it is. Then we don't need to check in the implementation.

Reply:
This prop is coming from the `props.classes` so it can potentially be `undefined`, that's why it is optional and we have the check. Note that, for the components that uses emotion, the `classes` prop is not augmented by the `styled` utility, that's why it can be undefined, if the developers does not have anything set for the prop on the component.

Reply:
> This prop is coming from the props.classes so it can potentially be undefined, that's why it is optional and we have the check.

So it can be `undefined` which is not the same (semantically) as being optional. Since we can express this at a type-level I'd argue we should change it to `classes: Record<string, string> | undefined`.

It seems to me that `composeClasses(slots, getUtilityClasses)` is wrong because we forgot to pass `classes`.

Reply:
Agree, let's add `classes: Record<string, string> | undefined` as type for it 👍

Reply:
Fixed in [`1fee388` (#24396)](https://github.com/mui-org/material-ui/pull/24396/commits/1fee38806c5f449148bad94798ddfa7b356d57da)